### DISCUSSION TITLE
Nuxt 3 + Axios

### DISCUSSION THREAD

**Discussion by kelvinmatheus** [nuxt/nuxt#18384] [Questions]
Good morning guys,

Does Nuxt 3 have support to Axios? 

If yes, do you guys have any documentation/example on how to install it?
If not, do we have an alternative with documentation/example?

Thanks for your time and attention!

**Comment by danielroe** [comment]
The best solution for Nuxt 3 is `$fetch()` (see [docs](https://v3.nuxtjs.org/docs/usage/data-fetching)) but you should be able to use axios by simply installing and importing it. Many of the convenience features of [`@nuxtjs/axios`](https://github.com/nuxt-community/axios-module/) are now present in the built-in `$fetch` helper.

**Reply by kelvinmatheus** [reply]
Thanks for the help. We are going to take a look on the documentation above!

**Reply by kelvinmatheus** [reply]
Thanks @danielroe, the `$fetch` has helped us. Do you guys have the intention to have the `[@nuxtjs/axios](https://github.com/nuxt-community/axios-module/)` module also implemented to Nuxt 3?

**Reply by danielroe** [reply]
Yes, we do intend to do so ☺️

**Reply by KahrLu** [reply]
> Yes, we do intend to do so ☺️

Hi @danielroe , I've been looking forward to what you said to make nuxt / Axios support vue3, But recently I saw that the document homepage of nuxt / Axios has guided developers to use $fetch. Will nuxt officials no longer consider upgrading this module.
If so, a little disappointed, we are going to use nuxt3 in enterprise projects. If we use ohmyfetch now, I don't believe we can convince most people. At least keep them on the same track... ...
please see more [#4514](https://github.com/nuxt/framework/discussions/4514)

**Reply by pi0** [reply]
Just to also clarify in this discussion on top of nuxt/nuxt#16605. Using `$fetch`/ohmyfetch is not enforced. You can use [https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API] in all environments of Nuxt. Also directly using axios library.

**Comment by pi0** [comment]
Let's continue discussion from one place ~> nuxt/nuxt#16605
